.github/workflows/codeql-analysis.yml

```diff
@@ -0,0 +1,72 @@
+# For most projects, this workflow file will not need changing; you simply need
```

Comment:
Do you think we should have all the static analysis and security checks in a single file, i.e. "checks.yml" to have the workflows folder more readable? So the only workflows we would have would be "tests.yml" that simply runs all the tests and generates the coverage report, checks.yml that does all the checks and deploy.yml that builds the image and publishes it to the registry. 

The problem with this is that the checks.yml file grows too large as we have a lot of checks. For minimizing checks.yml we could use tox which can run all the all the checks with a single command if configured properly. Might be a bit overkill to use though.

Reply:
Yeah, I agree with you. Another advantage is that Actions doesnt have to rebuild a Python image with our source on it for every action. However, I'm then not sure if we can make it show up as separate checks like now:

<img width="988" alt="image" src="https://user-images.githubusercontent.com/24796206/169424891-f58110a2-adb1-4e4d-aab1-501a463e2dfb.png">